src/middlewared/middlewared/plugins/account.py

```diff
@@ -156,30 +156,31 @@ class Config:
         # e.g. /, /root, /mnt/tank/my-dataset, etc ;).
         new_homedir = False
         home_mode = data.pop('home_mode')
-        if data['home'] != '/nonexistent':
+        home_dir = data['home']
+        if home_dir and home_dir != '/nonexistent':
```

Comment:
All these changes (data['home'] -> home_dir), seems unnecessary now. Can we revert to the original, please?